api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantServiceIntegrationTests.java

```diff
@@ -0,0 +1,256 @@
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.verify;
+import static org.springframework.http.HttpStatus.BAD_REQUEST;
+import static org.springframework.http.HttpStatus.CREATED;
+import static org.springframework.http.HttpStatus.INTERNAL_SERVER_ERROR;
+import static org.springframework.http.HttpStatus.NOT_FOUND;
+import static uk.gov.dhsc.htbhf.claimant.testsupport.ClaimDTOTestDataFactory.aValidClaimDTO;
+import static uk.gov.dhsc.htbhf.claimant.testsupport.ClaimDTOTestDataFactory.aValidClaimDTOWithNoNullFields;
+import static uk.gov.dhsc.htbhf.claimant.testsupport.EligibilityResponseTestDataFactory.anEligibilityResponse;
+import static uk.gov.dhsc.htbhf.claimant.testsupport.PersonDTOTestDataFactory.aValidPerson;
+
+public class ClaimantServiceIntegrationTests extends AbstractIntegrationTest {
+
+    // Create a string 501 characters long
+    private static final String LONG_STRING = CharBuffer.allocate(501).toString().replace('\0', 'A');
+
+    @Autowired
+    ObjectMapper objectMapper;
+
+    @Autowired
+    ClaimantRepository claimantRepository;
+
+    @MockBean
+    RestTemplate restTemplateWithIdHeaders;
+
+    @AfterEach
+    void deleteAllClaimants() {
+        claimantRepository.deleteAll();
+    }
+
+    @Test
+    void shouldAcceptAndCreateANewValidClaimWithNoNullFields() {
+        shouldAcceptAndCreateValidClaim(aValidClaimDTOWithNoNullFields());
+    }
+
+    @Test
+    void shouldAcceptAndCreateANewValidClaimWithNullFields() {
+        shouldAcceptAndCreateValidClaim(aValidClaimDTO());
+    }
+
+    private void shouldAcceptAndCreateValidClaim(ClaimDTO claim) {
+        //Given
+        ResponseEntity<EligibilityResponse> eligibilityResponse = new ResponseEntity<>(anEligibilityResponse(), HttpStatus.OK);
+        given(restTemplateWithIdHeaders.postForEntity(anyString(), any(), eq(EligibilityResponse.class))).willReturn(eligibilityResponse);
+        //When
+        ResponseEntity<Void> response = restTemplate.exchange(buildRequestEntity(claim), Void.class);
+        //Then
+        assertThat(response.getStatusCode()).isEqualTo(CREATED);
+        assertClaimantPersistedSuccessfully(claim.getClaimant(), EligibilityStatus.ELIGIBLE);
+        verify(restTemplateWithIdHeaders).postForEntity("http://localhost:8100/v1/eligibility", aValidPerson(), EligibilityResponse.class);
+    }
+
+    private void assertClaimantPersistedSuccessfully(ClaimantDTO claimantDTO, EligibilityStatus eligibilityStatus) {
+        Iterable<Claimant> claims = claimantRepository.findAll();
+        assertThat(claims).hasSize(1);
+        Claimant persistedClaim = claims.iterator().next();
+        assertClaimantMatchesClaimantDTO(claimantDTO, persistedClaim);
+        assertThat(persistedClaim.getEligibilityStatus()).isEqualTo(eligibilityStatus);
+    }
+
+    @Test
+    void shouldFailWhenEligibilityServiceCallThrowsException() {
```

Comment:
This and the test below should probably be in ClaimantServiceIntegrationErrorTests

Reply:
As discussed, we'll keep it in here as that class mocks out the `ClaimService`.